tests/spread/general/big/rockcraft.yaml

```diff
@@ -38,5 +38,12 @@ parts:
   usrmerge-part:
     plugin: nil
     override-build: |
+      # This build script adds a file in bin/ - the symlink in the base should be preserved.
       mkdir ${CRAFT_PART_INSTALL}/bin
       touch ${CRAFT_PART_INSTALL}/bin/new_bin_file
+    overlay-script: |
+      # explicitly remove a base layer symlink and create a dir instead
+      cd ${CRAFT_OVERLAY}
+      rm lib32
```

Comment:
it would be interesting to do this in separate parts, i.e. `partOne` -> `rm /lib32`, and then `partTwo` -> `mkdir`.

Reply:
Do you mean interesting in a testing sense? We can do that. I don't think that would be the typical usecase though

Reply:
Yes, in a test.

Indeed it is not the most common case, but I believe it is a more "stressful" case to test your changes. 
It is true that all parts are squashed into a single layer, so in theory this would still work well.

Maybe the real stress test is to create a `.rock` dir, since we always add a 2nd layer into the ROCK, for this path. Given that these are different layers, it is a true test for the whiteout and opaque features.

Reply:
> Maybe the real stress test is to create a .rock dir, since we always add a 2nd layer into the ROCK, for this path. Given that these are different layers, it is a true test for the whiteout and opaque features.

They are in different layers, but there wouldn't be a whiteout/opaque dir because nothing was "removed", right? Did you have a specific test in mind?

Anyway, I tried to do the original check of a part removing `/libx32` and then another part (re)creating `/libx32` and it didn't work: https://github.com/canonical/rockcraft/actions/runs/3697829487/jobs/6263399207
I expected the overlay mount to "persist" between parts but from the log it looks like it's mounted and unmounted for each part, so the original suggestion is not possible to express today. Is this something that we would want to support at some point?

Reply:
It's interesting though that the spread test only failed on ubuntu-18.04 (so it passed 20.04, 22.04, fedora, etc)

Reply:
> a part removing /libx32 and then another part (re)creating /libx32 and it didn't work

interesting. I feel this should be possible, after all that's how people expect `overlay-*` to work -> you're making changes on the final overlay system (a la Dockerfile style).

wrt to testing this with different layers, you could try having a part that creates `/.rock` as a symlink to something, and then see what happens to the 2nd layer (it should preserve the symlink `/.rock` and still put the metadata file inside)

Reply:
For the first point (one part removes/other part re-adds) I also feel like it should be possible. I suggest we create a separate issue to address this in the future

As for the second point, your suggestion is useful because [the test failed](https://github.com/canonical/rockcraft/actions/runs/3703751156/jobs/6275668885) and I realize that I haven't made this clear enough: the current implementation only looks at the base layer (ubuntu or bare), and not at the "current" layer stack. So the `.rock` symlink that the part added was hidden by the `.rock` dir in the metadata layer...
My suggestion for this is to *also* create an issue to support this in the future, which we'll definitely need if/when we start having per-part layers. In the meantime I can update the new test to document the fact that this *doesn't* work. Is this acceptable?

Reply:
> For the first point (one part removes/other part re-adds) I also feel like it should be possible. I suggest we create a separate issue to address this in the future

ok. but please make it a priority (maybe for the next sprint @sergiusens ?) cause it's clearly a problem.

> My suggestion for this is to also create an issue to support this in the future, which we'll definitely need if/when we start having per-part layers. 

ok. same as above then. with less priority though, since we don't yet support multi-layers (the .rock is a special case, implicitly caused by us)

Ok. sounds like a plan.

Reply:
Added #166 and #167 , thanks!